Neos.Neos/Classes/Fusion/Cache/CacheTag.php

```diff
@@ -51,9 +51,18 @@ final public static function forNodeAggregateFromNode(Node $node): self
         );
     }
 
+    final public static function forNodeAggregateFromNodeWithoutWorkspace(Node $node): self
+    {
+        return self::forNodeAggregate(
+            $node->contentRepositoryId,
+            null,
+            $node->aggregateId
+        );
```

Comment:
...change this to sth like
```suggestion
        return new self(
            'Node_'
            . self::getHashForContentRepositoryId($contentRepositoryId)
            . '_' . $nodeAggregateId->value
        );
```
and we could reduce the amount of repetition, by introducing some private `fromParts` constructor, e.g.:
```php
return self::fromParts(self::NODE_PREFIX, self::hashed($contentRepositoryId), $nodeAggregateId->value);
```

that would also make it easier to change our internal pattern because I think:
`<prefix>_sha1(<workspace>@<cr>)_<aggregateId>` is not safe (because the `NodeAggregateId` could contain underscores for example – not currently, but it is being [requested](https://github.com/neos/neos-development-collection/issues/5111))

`<prefix>|sha1(<workspace>)|sha1(<cr>)|<aggregateId>` would make more sense to me – even though: I don't understand why we randomly(?) use sha1 for 2 of those?

Reply:
The sha1 is to reduce/fix the length of the key.

Reply:
Sure, but can we move this into a dedicated issue?

Reply:
Yes, sorry ;)

Reply:
created an issue because the sha1 bugged me while debugging: https://github.com/neos/neos-development-collection/issues/5164